src/rendervulkan.cpp

```diff
@@ -159,15 +159,12 @@ struct {
 	uint32_t DRMFormat;
 	VkFormat vkFormat;
 	VkFormat vkFormatSrgb;
-	bool bNeedsSwizzle; // swap the R and the B channels
 	bool bHasAlpha;
 } s_DRMVKFormatTable[] = {
-	{ DRM_FORMAT_ARGB8888, VK_FORMAT_B8G8R8A8_UNORM, VK_FORMAT_B8G8R8A8_SRGB, false, true },
-	{ DRM_FORMAT_XRGB8888, VK_FORMAT_B8G8R8A8_UNORM, VK_FORMAT_B8G8R8A8_SRGB, false, false },
-	{ DRM_FORMAT_ARGB8888, VK_FORMAT_R8G8B8A8_UNORM, VK_FORMAT_R8G8B8A8_SRGB, true, true },
```

Comment:
Hm I think some GPUs are missing `VK_FORMAT_B8G8R8A8_UNORM`

Reply:
It's a required format (see Table 67. Mandatory format support: 4 byte-sized components), so those drivers wouldn't be conformant.

Reply:
Hmm. I don't remember why, but I had to add back this line for cursors to work at some point…

Reply:
Maybe it's a required format but not for the usage we require with our compute shaders?

Reply:
c9e8634e13560850e1227b6b939e636b1d56828a, but the cursor now uses bgra8 since 76292f01cb38a4c14ad2318e6d4efae67a5f6a1c.

Reply:
> Maybe it's a required format but not for the usage we require with our compute shaders?

Yes indeed, STORAGE_BIT is not required.

Reply:
> Hmm. I don't remember why, but I had to add back this line for cursors to work at some point…

Just tested this embedded, everything works as expected.

Reply:
IIRC the issue only happened on Intel.

Reply:
Okay, I have no intel machine so I can't verify that.

Reply:
Intel ANV supports BGRA8 storage images now so this should be fine.

https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/10624